dask_cloudprovider/generic/vmcluster.py

```diff
@@ -43,7 +43,10 @@ async def destroy_vm(self):
         raise NotImplementedError("destroy_vm is a required method of the VMInterface")
 
     async def wait_for_scheduler(self):
-        _, address = self.address.split("://")
+        if getattr(self, "external_address", None):
```

Comment:
I would expect `external_address` to always exist so it would probably be more readable to do:

```suggestion
        if self.external_address:
```